sycl/doc/extensions/SubGroupAlgorithms/SYCL_INTEL_sub_group_algorithms.asciidoc

```diff
@@ -0,0 +1,177 @@
+
+==== Permutes and Shifts
+
+The permute sub-group functions perform arbitrary communication between pairs of work-items in a sub-group.  Common patterns -- such as shifting all values in a sub-group by a fixed number of work-items, or reversing the order of all values in a sub-group -- are exposed as specialized functions that may be accelerated in hardware.
+
+|===
+|Function|Description
+
+|+template <typename T> T permute(sub_group sg, T x, id<1> local_id)+
+|Exchange values of _x_ between work-items in the sub-group in an arbitrary pattern.  Returns the value of _x_ from the work-item with the specified id.  The value of _local_id_ must be between 0 and the sub-group size.
+
+|+template <typename T> T shift_left(sub_group sg, T x, uint32_t delta)+
+|Exchange values of _x_ between work-items in the sub-group via a shift.  Returns the value of _x_ from the work-item whose id is _delta_ larger than the calling work-item. The value returned when the result of id + _delta_ is greater than or equal to the sub-group size is undefined.  The value of _delta_ must be the same for all work-items in the sub-group.
+
+|+template <typename T> T shift_right(sub_group sg, T x, uint32_t delta)+
+|Exchange values of _x_ between work-items in the sub-group via a shift.  Returns the value of _x_ from the work-item whose id is _delta_ smaller than the calling work-item.  The value of returned when the result of id - _delta_ is less than zero is undefined.  The value of _delta_ must be the same for all work-items in the sub-group.
+
+|+template <typename T> T xor_permute(sub_group sg, T x, id<1> mask)+
+|Exchange pairs of values of _x_ between work-items in the sub-group.  Returns the value of _x_ from the work-item whose id is equal to the exclusive-or of the calling work-item's id and _mask_. _mask_ must be a compile-time constant value that is the same for all work-items in the sub-group.
+
+|+template <typename T> T reverse(sub_group sg, T x)+
+|Exchange values of _x_ between work-items in the sub-group so as to reverse their order.  The value returned on work-item +i+ is the value of _x_ from the work-item whose id is equal to the sub-group size - +i+.
+
+|+template <typename T, class Compare> T sort(sub_group sg, T x, Compare comp)+
+|Exchange values of _x_ between work-items in the sub-group so as to reflect their ordering by the binary comparison function object _comp_.  _comp_ must be one of the comparison function objects from the group library.
+
+|===
+
+==== Loads and Stores
+
+The load and store sub-group functions enable developers to assert that all work-items in a sub-group read/write from/to contiguous locations in memory.  Such operations can be mapped directly to SIMD operations (when sub-groups are executed in SIMD fashion).
+
+|===
+|Function|Description
+
+|+template <typename T, access::address_space Space> T load(sub_group sg, const multi_ptr<T,Space> src)+
```

Comment:
I guess `Space` cannot be private, right?

Reply:
You're right.